src/Nethermind/Nethermind.Optimism/OptimismReceiptDecoder.cs

```diff
@@ -0,0 +1,165 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Serialization.Rlp;
+
+namespace Nethermind.Optimism;
+
+public class OptimismReceiptDecoder : IRlpStreamDecoder<OptimismTxReceipt>
```

Comment:
Can't we just extend base decoder?

Reply:
I made it this way because I didn't want to change base decoder. Inheriting will require few non-trivial changes. If you think it makes sense I can do it, but imo security worth copy-pasting few functions

Reply:
The question is would receipts change in the future and will those changes need to be replicated on Optimism. If yes than type hierarchy would be better as if someone would change base receipts, this would immediately reflect here. It would make discoverability of places that need to change easier. It is a hard question that I don't have the answer to.